content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md

```diff
@@ -144,9 +144,66 @@ For more information on version skews, see:
 * Kubernetes [version and version-skew policy](/docs/setup/release/version-skew-policy/)
 * Kubeadm-specific [version skew policy](/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm/#version-skew-policy)
 
+{{< warning >}}
+Kubernetes has two different package repositories starting from August 2023.
+The Google-hosted repository is considered deprecated and we will stop
+publishing packages to that repository starting with Kubernetes 1.30.0.
+The community-hosted repositories are replacement for the Google-hosted
+repository. There are some important considerations for the community-hosted
+repositories:
+
+- We strongly recommend using the community-hosted repositories especially for
+  new setups. For existing setups, we recommend migrating to the community-hosted
+  repositories. The migration process is described in the following document: TBD.
+- The community-hosted repositories contain only packages starting with
+  Kubernetes 1.24.0. Earlier versions are available only in the Google-hosted
+  repository.
+- We have a different repository for each Kubernetes minor release.
+  Please take this into consideration when upgrading to the next minor release.
+
+For more information about community-hosted repositories, we recommend
+checking out the following document: TBD
+{{< /warning >}}
+
 {{< tabs name="k8s_install" >}}
 {{% tab name="Debian-based distributions" %}}
 
+### Community-hosted repositories
+
+1. Update the `apt` package index and install packages needed to use the Kubernetes `apt` repository:
+
+   ```shell
+   sudo apt-get update
+   sudo apt-get install -y apt-transport-https ca-certificates curl
+   ```
+
+2. Download the public signing key for community-hosted repositories. The same signing key is used for all repositories so you can disregard the version in the URL:
+
+   ```shell
+   curl -fsSL https://pkgs.k8s.io/core:/stable:/{{< param "version" >}}/deb/Release.key | sudo gpg --dearmor -o /etc/apt/keyrings/kubernetes-apt-keyring.gpg
+   ```
+
+3. Add the appropriate community-hosted Kubernetes `apt` repository. If you want to use Kubernetes version different than {{< param "version" >}},
+   replace {{< param "version" >}} with the desired minor version in the command below:
+
+   ```shell
+   echo 'deb https://pkgs.k8s.io/core:/stable:/{{< param "version" >}}/deb/ /' | sudo tee /etc/apt/sources.list.d/kubernetes.list
+   ```
+
+4. Update `apt` package index, install kubelet, kubeadm and kubectl, and pin their version:
+
+   ```shell
+   sudo apt-get update
+   sudo apt-get install -y kubelet kubeadm kubectl
+   sudo apt-mark hold kubelet kubeadm kubectl
+   ```
+{{< note >}}
+In releases older than Debian 12 and Ubuntu 22.04, `/etc/apt/keyrings` does not exist by default.
+You can create this directory if you need to, making it world-readable but writeable only by admins.
+{{< /note >}}
+
+### Google-hosted repository (Deprecated)
```

Comment:
> Should we change the name of the file within /etc/apt/keyrings? Eg to /etc/apt/keyrings/kubernetes-deprecated-apt-keyring.gpg

I'd avoid doing this so that we have universal file location for the old key for all users in case we ever need to get back to it, e.g. to remove it if needed. I used a new location for the new key, I think that's kind of a better approach.